File: src/main/java/com/fengsheng/handler/display_record_tos.java
package com.fengsheng.handler;

import com.fengsheng.HumanPlayer;
import com.fengsheng.protos.Fengsheng;
import com.google.protobuf.GeneratedMessageV3;
import org.apache.log4j.Logger;

public class display_record_tos implements ProtoHandler {
    private static final Logger log = Logger.getLogger(display_record_tos.class);

    @Override
    public void handle(HumanPlayer player, GeneratedMessageV3 message) {
        if (player.getGame() != null) {
            log.error("player is already in a room");
            return;
        }
        var pb = (Fengsheng.display_record_tos) message;
        player.loadRecord(pb.getVersion(), pb.getRecordId());
    }
}

File: src/main/kotlin/handler/DisplayRecordTos.kt
package com.fengsheng.handler

import com.fengsheng.HumanPlayer
import com.fengsheng.protos.Fengsheng
import com.google.protobuf.GeneratedMessage
import org.apache.logging.log4j.kotlin.logger

class DisplayRecordTos : ProtoHandler {
    override fun handle(player: HumanPlayer, message: GeneratedMessage) {
        if (player.game != null || player.isLoadingRecord) {
            logger.error("player is already in a room")
            player.sendErrorMessage("你已经在房间里了")
            return
        }
        val pb = message as Fengsheng.display_record_tos
        player.loadRecord(pb.version, pb.recordId, pb.skipCount)
    }
}
